BasicSample/Assets/GrabAndThrow/Scripts/GrabAndThrowBehavior.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+using UnityEngine.XR;
+
+namespace Microsoft.MixedReality.OpenXR.BasicSample
+{
+	[RequireComponent(typeof(Rigidbody))]
+	public class GrabAndThrowBehavior : MonoBehaviour
+	{
+		private const float PICKABLE_RANGE = 0.5f;
+
+		[SerializeField]
+		public BoxCollider PickupDetectionCollider;
+
+		private bool[] m_wasDeviceTapped = { false, false };
+		private Rigidbody rigidBody;
+
+		private XRNode? handHeldBy = null;
+		private Transform handSpace;
+		private Vector3 positionInHandSpace;
+		private Quaternion rotationInHandSpace;
+
+		private void Awake()
+		{
+			rigidBody = GetComponent<Rigidbody>();
+			GameObject handSpaceGameObject = new GameObject("Hand Space");
+			handSpace = Instantiate(handSpaceGameObject, Vector3.zero, Quaternion.identity).transform;
+		}
+
+		// Update is called once per frame
+		private void Update()
+		{
+			// Repeat this update once for each hand
+			for (int i = 0; i < 2; i++)
+			{
+				XRNode handNode = (i == 0) ? XRNode.RightHand : XRNode.LeftHand;
+				InputDevice device = InputDevices.GetDeviceAtXRNode(handNode);
+				bool deviceHasData = device.TryGetFeatureValue(CommonUsages.isTracked, out bool deviceIsTracked);
+				deviceHasData &= device.TryGetFeatureValue(CommonUsages.primaryButton, out bool isDeviceTapped);
+				deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceVelocity, out Vector3 deviceVelocity);
+				deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceAngularVelocity, out Vector3 deviceAngularVelocity);
+				deviceHasData &= device.TryGetFeatureValue(CommonUsages.devicePosition, out Vector3 devicePosition);
+				deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceRotation, out Quaternion deviceRotation);
+
+				if (deviceHasData && deviceIsTracked)
+				{
+					handSpace.SetPositionAndRotation(devicePosition, deviceRotation);
+
+					// Check if we need to grab the object
+					if (handNode != handHeldBy && isDeviceTapped && !m_wasDeviceTapped[i])
+					{
+						Vector3 positionInVolumeCoordinates = PickupDetectionCollider.transform.InverseTransformPoint(devicePosition);
+						bool canBePickedUp = false;
+						if (Mathf.Abs(positionInVolumeCoordinates.x) < PICKABLE_RANGE &&
```

Comment:
Is it possible to use "https://docs.unity3d.com/ScriptReference/Collider.ClosestPoint.html" to make this test instead of using point distances?

Reply:
Or maybe this `Physics.OverlapSphere` function can also helpful, which gives multiple colliders given a position and radius.

Reply:
Collider.ClosestPoint worked great; thanks for the tip!